server/plugin.go

```diff
@@ -22,6 +22,12 @@ func (p *Plugin) FileWillBeUploaded(c *plugin.Context, info *model.FileInfo, fil
 	config := p.getConfiguration()
 
 	av := clamd.NewClamd("tcp://" + config.ClamavHostPort)
+	err := av.Ping()
```

Comment:
Could we simplify this and just return the nice error message on line 35? Do we need an extra round trip the the AV server?

Reply:
If we directly return the server unreachable error message in line 35, it will become generic for all error scenarios of av.ScanStream() function. So, even if server is up and running but ScanStream() threw error while reading the stream, it will show as server unreachable only. 

Yes, we should try to avoid the round trip to the AV server. In that case, we would need an extra check inside if condition of line 33, something like 

```
if strings.Contains(err.Error(), "connection refused"){
		// return error message
}
```

Let me know your views.

Reply:
From a user's point of view. The detail of the message isn't increased by saying the server was unreachable. The administrator, to debug the issue, would still have to look in the logs and contact the admin for assistance.

Reply:
Thanks for your valuable feedback. I have made the changes accordingly so that now we don't do an extra ping and just throw the custom error inside ScanStream() function itself.

Let me know if that's what is expected.